charts/scalar-manager/templates/scalar-manager/deployment.yaml

```
@@ -54,15 +111,131 @@ spec:
+            - name: TLS_OUTBOUND_GRAFANA_ENABLED
+              value: "{{ .Values.scalarManager.tls.upstream.grafana.enabled }}"
+            - name: TLS_OUTBOUND_GRAFANA_CA_ROOT_CERT_PATH
+              value: "/tls/upstream/grafana/certs/ca.crt"
+            {{- if .Values.scalarManager.tls.upstream.grafana.overrideAuthority }}
+            - name: TLS_OUTBOUND_GRAFANA_OVERRIDE_AUTHORITY
+              value: "{{ .Values.scalarManager.tls.upstream.grafana.overrideAuthority }}"
+            {{- end }}
+            {{- end }}
           ports:
-            - containerPort: 3000
+            - containerPort: {{ .Values.scalarManager.web.service.ports.web.targetPort }}
           imagePullPolicy: {{ .Values.scalarManager.web.image.pullPolicy }}
+          volumeMounts:
+            {{- if .Values.scalarManager.tls.downstream.enabled }}
+            - name: scalar-manager-tls-volume
+              mountPath: /tls/scalar-manager/certs
+            {{- end }}
+            {{- if .Values.scalarManager.tls.downstream.caRootCertSecret }}
+            - name: scalar-manager-tls-downstream-ca-volume
+              mountPath: /tls/ca/certs
+            {{- end }}
+            {{- if and .Values.scalarManager.tls.upstream.grafana.enabled .Values.scalarManager.tls.upstream.grafana.caRootCertSecret }}
+            - name: scalar-manager-tls-upstream-grafana-volume
+              mountPath: /tls/upstream/grafana/certs
+            {{- end }}
           securityContext:
             {{- toYaml .Values.scalarManager.securityContext | nindent 12 }}
       volumes:
         - name: api-application-properties-volume
           configMap:
             name: {{ include "scalar-manager.fullname" . }}-api-application-properties
+        {{- if and .Values.scalarManager.tls.downstream.enabled (not .Values.scalarManager.tls.certManager.enabled) }}
+          {{- if or (not .Values.scalarManager.tls.downstream.certChainSecret) (not .Values.scalarManager.tls.downstream.privateKeySecret) }}
+            {{- fail "When using manual downstream TLS, both scalarManager.tls.downstream.certChainSecret and scalarManager.tls.downstream.privateKeySecret must be set." -}}
+          {{- end }}
+        - name: scalar-manager-tls-volume
+          projected:
+            sources:
+              - secret:
+                  name: {{ .Values.scalarManager.tls.downstream.certChainSecret }}
+                  items:
+                    - key: tls.crt
+                      path: tls.crt
+              - secret:
+                  name: {{ .Values.scalarManager.tls.downstream.privateKeySecret }}
+                  items:
+                    - key: tls.key
+                      path: tls.key
+        {{- end }}
+        {{- if and .Values.scalarManager.tls.downstream.enabled .Values.scalarManager.tls.certManager.enabled }}
+        - name: scalar-manager-tls-volume
+          secret:
+            secretName: {{ include "scalar-manager.fullname" . }}-tls-cert
```

```suggestion
        - name: scalar-manager-tls-volume
          secret:
            secretName: {{ include "scalar-manager.fullname" . }}-tls-cert
            items:
              - key: tls.key
                path: tls.key
              - key: tls.crt
                path: tls.crt
```
I think cert-manager generates the secret with keys `tls.key` and `tls.crt`. In other words, those key names are guaranteed by cert-manager.

However, I think it might be better to check the key names here for safety, just in case.

It can also keep the consistency between other volume mount configurations in this chart.

What do you think?